sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -671,6 +673,33 @@ trait GpuHashJoin extends GpuExec {
       new ConditionalHashJoinIterator(spillableBuiltBatch, boundBuildKeys, lazyStream,
         boundStreamKeys, streamedPlan.output, compiledCondition,
         realTarget, joinType, buildSide, compareNullsEqual, spillCallback, opTime, joinTime)
+    } else if (joinType.isInstanceOf[ExistenceJoin]) {
+      new Iterator[ColumnarBatch]() {
```

Comment:
Nit: I think this is enough code for this iterator to have a class for it, or at least a private method that creates this iterator, to make this `val joinIterator = if (...` block more compact and readable.

Reply:
agreed. This is just a WIP prototype code with everything leaking but passing the test, which I wanted to show you to get a confirmation whether this is close to the solution you suggest.